dask/dataframe/shuffle.py

```diff
@@ -64,7 +64,8 @@ def set_index(df, index, npartitions=None, shuffle=None, compute=False,
         divisions, sizes, mins, maxes = base.compute(divisions, sizes, mins, maxes)
         divisions = divisions.tolist()
 
-        if repartition:
+        empty_dataframe_detected = all(pd.isnull(d) for d in divisions)
```

Comment:
`pd.isnull` operatres on arrays too, so this could be `pd.isnull(divisions).all()`